src/host/srvinit.cpp

```diff
@@ -388,6 +391,22 @@ try
         return E_NOT_SET;
     }
 
+    // Capture handle to the inbox process into a unique handle holder.
+    g.handoffInboxConsoleHandle.reset(hostProcessHandle);
+
+    // Set up a threadpool waiter to shutdown everything if the inbox process disappears.
+    g.handoffInboxConsoleExitWait.reset(CreateThreadpoolWait(
+        [](PTP_CALLBACK_INSTANCE /*callbackInstance*/, PVOID /*context*/, PTP_WAIT /*wait*/, TP_WAIT_RESULT /*waitResult*/) noexcept {
+            ServiceLocator::RundownAndExit(E_APPLICATION_MANAGER_NOT_RUNNING);
+        },
+        nullptr,
+        nullptr));
+
+    SetThreadpoolWait(g.handoffInboxConsoleExitWait.get(), g.handoffInboxConsoleHandle.get(), nullptr);

```

Comment:
I've read the documentation, but I don't quite get how it works...
Is this functionally the same as `RegisterWaitForSingleObject`?

Reply:
I think you're right. I think it's similar functionality to that function, just perhaps with different configuration options available.